### ISSUE TITLE
How to set MAX_LABEL in custom dataset

### ISSUE THREAD

**Issue by taosean** [nwojke/cosine_metric_learning#49]
Hi, @nwojke , I'm trying to train on my own dataset, I created the dataset just as Market 1501 format (naming format). Now my training dataset has 500 identities, indexing from 0001 to 0500. 

Now I am confused about how to set the `MAX_LABEL`. I see [here](https://github.com/nwojke/cosine_metric_learning/blob/eda0daaa5462c61ac44553f2151070fd7e316cc8/datasets/market1501.py#L9) `MAX_LABEL` is set to 1501, however, the maximum id number of images inside Market1501/bounding_box_train is 1500.

Could you tell me how to set the `MAX_LABEL` in my case, and what's the purpose of this value?

Thanks!

**Comment by nwojke** [comment]
The ``MAX_LABEL`` is used to set the number of classes [here](https://github.com/nwojke/cosine_metric_learning/blob/eda0daaa5462c61ac44553f2151070fd7e316cc8/train_market1501.py#L62). So ``MAX_LABEL + 1`` specifies the number of units in the classification layer.

If you have labels up to 500, ``MAX_LABEL`` should be 500 and ``num_classes`` should be 501. If the maximum label in Market 1501 is 1500, then we have an empty class in our code.

**Comment by taosean** [comment]
@nwojke 
Thank you for your quick response. I understand that `num_classses` is the length of one-hot encoding vector. I wonder, if we have several empty class, will it affect the result?

And I have another question. 
Since I'm training on my own dataset, and I will get an 128-d vector from each image, do you have any metric to evaluate distance between these 128-d vectors directly, not using Market1501 or MARS evaluation software?

**Comment by nwojke** [comment]
Empty classes will occupy _some_ space in the feature space and samples which fall within this area are going to receive a penalty. Since the scale parameter is shared between all classes, the optimizer cannot collapse this empty class region into a single point. However, if there is more than one empty class it can push all empty classes into the same region, isolated from remaining classes. Based on this reasoning, but without having performed any experiments, I don't think empty classes have a large impact on performance.

The repository contains code for mAP and CMC metrics  (e.g. [here](https://github.com/nwojke/cosine_metric_learning/blob/master/metrics.py#L166)). I remember that I have compared the results to the Market 1501 SDK and values have been very close. The code is used to measure performance during evaluation, but you could also write your own evaluation script based on them. I am not sure how much support I can give though, since I haven't touched  the code in a while and I would have to read into it again as well.

**Comment by joos2010kj** [comment]
Hello,

Could I ask why mars.MAX_LABEL is 1500?
